lib/reverse_sentence.rb

```diff
@@ -1,6 +1,24 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n) the method below does not sort in place (.split makes a copy of the original string). The array gets bigger as the input is entered
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return "" if my_sentence.nil? || my_sentence.empty?
+  new_sentence = my_sentence.split(' ') #makes copy
```

Comment:
Consider splitting with regex like `\s+`  That might split and handle it with more than one space between words.